### ISSUE TITLE
Has anyone had a good trained model

### ISSUE THREAD

**Issue by retoriax** [microsoft/KBLaM#75]
Has anyone had got a good trained model and can share what dataset, model and training parameters/flags they used for training and eval?
I cant get good scores with kblam using Llama-3-8B-Instruct, text-embedding-3-large and the synthetic dataset.
Maybe also share what the final train loss and step count was.

**Comment by retoriax** [comment]
Thank you very much @xidulu 
I trained for 30h on a A100 and got good results.
{'rouge1': 0.7812977846708903, 'rouge2': 0.6080092027443471, 'rougeL': 0.7432360570209307, 'rougeLsum': 0.7433776824138043, 'bert_score_precision': 0.9007512772083283, 'bert_score_recall': 0.9034255331754685, 'bert_score_f1': 0.9011788907647132, 'mem_cost': 21168652288}
My train_loss was at 0.53 it took 5k steps.

I used kb_layer_frequency 2, with ada-002 and Llama 8B.
I created my own training dataset from scatch off of wikipedia entries.

Can anyone explain how --kb-scale-factor works exactly? 
I can only guess for good values but is there a better approach?

**Comment by xidulu** [comment]
@retoriax 
The metric number looks not bad at all!!!!

> I created my own training dataset from scatch off of wikipedia entries.

So glad to hear KBLaM works on users' customized dataset !!!

> Can anyone explain how --kb-scale-factor works exactly?

See this paragraph in the paper for a detailed introduction, kb-scale-factor controls the magnitude of the contribution from the KB's value embeddings, it is introduced to make sure the model can work with arbitrary sizes of KB.

<img width="841" height="271" alt="Image" src="https://github.com/user-attachments/assets/7d452248-3b8f-4603-958b-14e0a9edc3c5" />

**Comment by vpgits** [comment]
Hi @retoriax , 
were you successful with training Llama 3.2 1B?

**Comment by retoriax** [comment]
Hi @vpgits,
unfortunately I wasn’t successful.

After xidulu mentioned “Also I have never tried running the model with Llama 1B myself, which per memory during the rebuttal, has pretty concerning performance, the 8B model may have much better performance.”
I immediately switched to the 8B models instead.

I hope you’ll have more success with the 1B models.

**Comment by vpgits** [comment]
I had to completely rewrite the modelling llama class to account for the new RoPE embeddings in llama models, , they have significantly restructured the api. And then had to rewrite the entire codebase to mimic the original repo. However i keep running into issues. I think my implementation might be wrong to be honest, but I dont see why a 1B param model should fail.
